### PR TITLE
Reduce findings span - part 2

### PR DESCRIPTION
**Pull request by p4p3r**
The purpose of this PR is to reduce the number of lines of the findings reported by rules that have shown to be matching on average very large ranges.

Marked as draft while the impact is assessed.

### CODE REVIEW

File: python/django/security/audit/django-rest-framework/missing-throttle-config.yaml
Comment (0xDC0DE) [inline python/django/security/audit/django-rest-framework/missing-throttle-config.yaml]: Should this be `pattern-inside` ?

**Review by 0xDC0DE — APPROVED** [review]

File: python/django/security/audit/django-rest-framework/missing-throttle-config.yaml
Comment (p4p3r) [reply python/django/security/audit/django-rest-framework/missing-throttle-config.yaml]: The rules works either way.  Do you think `pattern-inside` would be more appropriate?

Comment (0xDC0DE) [reply python/django/security/audit/django-rest-framework/missing-throttle-config.yaml]: It feels counter-intuitive.
If the match needs to be `REST_FRAMEWORK = ...` and `REST_FRAMEWORK` at the same time, the clauses seems either conflicting (can not be both at the same time) or redundant (one is more restrictive than the other).

Comment (p4p3r) [reply python/django/security/audit/django-rest-framework/missing-throttle-config.yaml]: I see why `pattern-inside` would make more sense, as I'm using patterns to step-by-step drill down to a specific part "keyword".
Addressed in 72b013cb.

File: typescript/react/security/audit/react-styled-components-injection.jsx
Comment (0xDC0DE) [inline typescript/react/security/audit/react-styled-components-injection.jsx]: I've also come across cases like this. I wonder if there is a better way to do this, instead of making the code a single line. Should we have smarter test syntax?

**Review by 0xDC0DE — APPROVED** [review]

File: typescript/react/security/audit/react-styled-components-injection.jsx
Comment (p4p3r) [reply typescript/react/security/audit/react-styled-components-injection.jsx]: Oh actually it works fine with comments inside the curly braces! Fixed in ab710c844482bbf3afa24c1ee22862bc356e3384

**Comment by colleend** [comment]
are we waiting on someone before this PR is merge-able?

**Comment by p4p3r** [comment]
> are we waiting on someone before this PR is merge-able?

@colleend yes, for both this and https://github.com/returntocorp/semgrep-rules/pull/2081 we are waiting on some work from the app team.

**Comment by colleend** [comment]
got it, I'll note that in support queue 👍

**Pull request merged**
